pkg/routeagent/controllers/route/driver.go

```diff
@@ -3,32 +3,57 @@ package route
 import (
 	"fmt"
 	"io/ioutil"
+	"net"
 
 	"github.com/vishvananda/netlink"
 	"k8s.io/klog"
+
+	"github.com/submariner-io/submariner/pkg/util"
 )
 
-const FlannelInterface = "flannel.1"
+func interfaceExists(iface string) bool {
+	_, err := netlink.LinkByName(iface)
+	return err == nil
+}
 
-func isFlannelCNI() error {
-	_, err := netlink.LinkByName(FlannelInterface)
-	if err != nil {
-		return fmt.Errorf("error retrieving link by name %s: %v", FlannelInterface, err)
+func discoverCNIInterface(clusterCIDR string) string {
+	cniInterfaceList := []string{
+		"weave",     // Weave
+		"tun0",      // OpenShift
+		"flannel.1", // Flannel
 	}
 
-	// FlannelInterface exists, return nil
-	return nil
-}
+	for _, iface := range cniInterfaceList {
+		if interfaceExists(iface) {
+			klog.Infof("Found CNI interface [%s] on the Host", iface)
+			ipv4Addr, err := util.GetIPv4AddressOnInterface(iface)
+			if err != nil {
+				klog.Errorf("error reading IPv4 address on CNI interface [%s]: %v", iface, err)
+				return ""
+			}
+			address := net.ParseIP(ipv4Addr)
 
-func toggleCNISpecificConfiguration() error {
-	// When using Flannel CNI, reverse path filtering has to be configured with loose mode on the FlannelInterface
-	if isFlannelCNI() == nil {
-		err := ioutil.WriteFile("/proc/sys/net/ipv4/conf/"+FlannelInterface+"/rp_filter", []byte("2"), 0644)
-		if err != nil {
-			return fmt.Errorf("unable to update rp_filter for "+FlannelInterface+", err: %s", err)
-		} else {
-			klog.Info("Successfully configured rp_filter to loose mode(2) on " + FlannelInterface)
+			_, clusterNetwork, err := net.ParseCIDR(clusterCIDR)
+			if err != nil {
+				klog.Errorf("Unable to ParseCIDR [%s] : %v\n", clusterCIDR, err)
+				return ""
+			}
+
+			// Verify that iface has an address from cluster CIDR
+			if clusterNetwork.Contains(address) {
+				return iface
+			}
 		}
 	}
+	return ""
+}
+
+func toggleCNISpecificConfiguration(iface string) error {
+	err := ioutil.WriteFile("/proc/sys/net/ipv4/conf/"+iface+"/rp_filter", []byte("2"), 0644)
+	if err != nil {
+		return fmt.Errorf("unable to update rp_filter for "+iface+", err: %s", err)
```

Comment:
nit/idea, no action required: May be in such error case we can also explain "hostNetwork to remote podIP/clusterIPs won't work" ? ..

Reply:
ack thanks